File: day_3/python/old/wordcount.py
wordsFromFileRDD = sc.textFile("file:///opt/sample-data/pg4300.txt")

wordRDD = wordsFromFileRDD.flatMap(lambda l: l.split(" "))
mappedWordRDD = wordRDD.map(lambda w: (w,1))
#reduce the word by counting occurence of unique word
reducedWordRDD = mappedWordRDD.reduceByKey(lambda a, b: a + b)
#sort the result by the word
sortedRDD = reducedWordRDD.sortByKey()
#we now want to tab delimit the fields of the output
outputRDD = sortedRDD.map(lambda x: x[0] + "\t" + str(x[1]))
#write as textFile, sequencefile
outputRDD.saveAsTextFile("/user/cloudera/output/py_out/pg4300_tab")


outputRDD.map(lambda x: (None, x)).saveAsSequenceFile("/user/cloudera/output/py_out/pg4300_tab_seq")

File: spark1-scala/flight_records/wordcount.scala
val wordsFromFileRDD = sc.textFile("file:///opt/sample-data/pg4300.txt")
//val wordRDD = wordsFromFileRDD.flatMap(l => l.split(" "))
//val outputRDD = wordRDD.countByValue

val wordRDD = wordsFromFileRDD.flatMap(l => l.split(" "))
val mappedWordRDD = wordRDD.map(w => (w,1))
//reduce the word by counting occurence of unique word
val reducedWordRDD = mappedWordRDD.reduceByKey((a, b) => a+b)
//we now want to tab delimit the fields of the output
val outputRDD = reducedWordRDD.map(x => x._1 + "\t" + x._2)
//sort the result by the word
val sortecRDD = outputRDD.sortBy(a => a)
//write as textFile, sequencefile
sortecRDD.saveAsTextFile("/user/cloudera/output/pg4300_tab")


sortecRDD.map(l => ("", l)).saveAsSequenceFile("/user/cloudera/output/pg4300_tab_seq")